src/student/student.controller.ts

```diff
@@ -1,34 +1,130 @@
+import { User } from './entities/user.entity';
+import {
+  BoolValues,
+  ExpectedContractType,
+  ExpectedTypeWork,
+  FilterSettings,
+  GetPaginatedListOfUser,
+  Status,
+} from 'types';
+import { ParseFilterIntPipe } from 'src/pipes/parse-filterInt.pipe';
+import { ParseFilterBooleanPipe } from 'src/pipes/parse-filterBoolean.pipe';
+import { ParseExpectedContractTypePipe } from 'src/pipes/parse-expectedContractType.pipe';
+import { ParseExpectedTypeWorkPipe } from 'src/pipes/parse-expectedTypeWork.pipe';
+import { ParseScorePipe } from 'src/pipes/parse-score.pipe';
+import { RoleGuard } from 'src/auth/role.guard';
+import { Roles } from 'src/decorators/roles.decorator';
+import { Role } from './interfaces/user';
+import { ParseStatusPipe } from 'src/pipes/parse-status.pipe';
+import { ParseFilterStringPipe } from 'src/pipes/parse-filterString.pipe';
 import { UpdateStudentDto } from './dto/update-student.dto';
 
 @Controller('student')
 export class StudentController {
   constructor(private readonly studentService: StudentService) {}
 
   @Post()
-  create(@Body() createStudentDto: CreateStudentDto) {
+  async create(@Body() createStudentDto: CreateStudentDto) {
     return this.studentService.create(createStudentDto);
   }
 
-  @Get()
-  findAll() {
-    return this.studentService.findAll();
+  @Patch('/password')
+  @UseGuards(AuthGuard('jwt'))
+  async changePassword(
+    @UserObj() user: User,
+    @Body() { password }: { password: string },
+  ) {
+    return await this.studentService.changePassword(user, password);
   }
 
-  @Get(':id')
-  findOne(@Param('id') id: string) {
-    return this.studentService.findOne(+id);
+  @Roles(Role.STUDENT)
+  @UseGuards(AuthGuard('jwt'), RoleGuard)
+  @Patch('/hired')
+  async changeStatusToHiredAndDeactivateAccount(@UserObj() user: User) {
+    return await this.studentService.changeStatusToHiredAndDeactivateAccount(
+      user,
+    );
   }
 
-  @Patch(':id')
-  update(@Param('id') id: string, @Body() updateStudentDto: UpdateStudentDto) {
-    return this.studentService.update(+id, updateStudentDto);
+  @Roles(Role.STUDENT)
+  @UseGuards(AuthGuard('jwt'), RoleGuard)
+  @Patch('/update')
+  async updateAccount(
+    @UserObj() user: User,
+    @Body() changes: UpdateStudentDto,
+  ) {
+    return await this.studentService.changeStudentData(user, changes);
   }
 
-  @Delete(':id')
-  remove(@Param('id') id: string) {
-    return this.studentService.remove(+id);
+  @Roles(Role.STUDENT)
+  @UseGuards(AuthGuard('jwt'), RoleGuard)
+  @Get('/profile')
+  async getStudentProfile(@UserObj() user: User) {
+    return await this.studentService.getStudentProfile(user);
+  }
+
+  @Roles(Role.HR)
+  @UseGuards(AuthGuard('jwt'), RoleGuard)
+  @Get(
+    '/filtered/:perPage/:pageNumber/:status/:firstName/:lastName/:courseCompletion/:courseEngagement/:projectDegree/:teamProjectDegree/:expectedTypeWork/:expectedContractType/:minNetSalary/:maxNetSalary/:canTakeApprenticeship/:monthsOfCommercialExp/:email',
```

Comment:
Można też to ograć za pomocą Query, byłoby trochę ładniej.

Reply:
Zrobione. Niestety Query sprawił, że url na froncie wygląda dużo gorzej, dlatego stworzyłem funkcję, która generuje zapytanie 👍